src/Repositories/AbstractJsonApiDoctrineRepository.php

```diff
@@ -147,44 +147,11 @@ public function findByArray(
     }
 
     /**
-     * Updates or creates an Entity.
+     * Saves an Entity to the database.
      *
      * @phpstan-param Entity $entity
      */
-    public function createOrUpdate(object $entity): void
-    {
-        $this->isValidEntity($entity);
-
-        $e = $this->findById($entity->id());
-
-        if ($e == null) {
-            $this->add($entity);
-        } else {
-            $this->update($entity);
-        }
-    }
-
-    /**
-     * Update an existing Entity.
-     *
-     * @phpstan-param Entity $entity
-     */
-    public function update(object $entity, bool $mute = false): void
-    {
-        $this->isValidEntity($entity);
-        $this->em->merge($entity);
-
-        if (!$mute) {
-            $this->em->flush();
-        }
-    }
-
-    /**
-     * Add a new Entity to the database.
-     *
-     * @phpstan-param Entity $entity
-     */
-    public function add(object $entity, bool $mute = false): void
+    public function save(object $entity, bool $mute = false): void
     {
         $this->isValidEntity($entity);
```

Comment:
Totally unrelated, but because we are unlikely to be in here again soon, can we fix this?

- `is` implies it will return a boolean
- `assert` implies it will throw if invalid


```suggestion
        $this->assertValidEntity($entity);
```